### DISCUSSION TITLE
RFE: Insulin Pens - Changing insulin types

### DISCUSSION THREAD

**Discussion by Loquacity** [NightscoutFoundation/xDrip#3382] [Q&A]
After digging through the code, it appears that it is not possible to edit the JSON file that stores the insulin types, due to Android security permissions. Switching this to an issue and a request for a feature: possibly just adding an option to the Insulin Pens menu to forget a saved device. Changing the insulin type in a pen is trivial, it should also be trivial to reset the insulin pen (or 'forget' it so it can set up again) in the xDrip UI as well.

Original discussion text follows:

### Discussed in https://github.com/NightscoutFoundation/xDrip/discussions/3322

<div type='discussions-op-text'>

<sup>Originally posted by **Loquacity** January 29, 2024</sup>
Hi everyone,

I'm using two 'smart' insulin pens, both are NovoPen 6, and both are currently being used with my basal insulin (Levemir, although they're saved in xDrip as Lantus, since Levemir was not an available insulin type at the time). With the release of my bolus insulin (Fiasp) in a penfill, I am hoping to switch to using one smart pen for bolus instead, and the other for basal. I thought that if I could get xDrip to 'forget' my pens, then I could re-add them on the next dose, and be able to set them up with the correct insulin, but I can't see a way to do that. Is there some way to either have xDrip 'forget' the pen, or a way to manually change which type of insulin the pen is using? Thanks for all the help!</div>

**Comment by Navid200** [comment]
You can bump a discussion by posting in it.  You cannot do that with an issue.

At this point, we need a developer to look at this and tell us what they can do with it or if anything can be done.
If yes, we can make it an issue so that the developers can prioritize.
But, if there is nothing we can do about it, there is no point in having an open issue about it.

While this is a discussion, as I said already, you can bump it by posting in it.  Then, it will become the first discussion.  So, it is more likely to get the attention of a developer.
If it is an issue, you will never be able to move it up.

**Comment by Navid200** [comment]
If you look at the oldest open issue, you will see that currently, it is from 2017. That's almost 7 years ago.
So, just because something is an open issue does not guarantee it will be done.

Yet, if everything is done with issues, there will be too many of them and none of them will get any attention.
Even the ones that may have a chance of being addressed will be buried under all sorts of requests.

**Comment by aussiexjohn** [comment]
Hi Loquacity,
Not very intuitive, but changing insulin type for future doses can be done from the graph.  Select a dose, and it gives an option.  Your other thread https://github.com/NightscoutFoundation/xDrip/issues/3322 included a pointer to the original announcement https://github.com/NightscoutFoundation/xDrip/issues/2074 which mentioned this (in the opening post).  
J.
